L1Trigger/L1THGCal/interface/backend/HGCalShowerShape.h

```diff
@@ -72,6 +78,20 @@ class HGCalShowerShape {
       X_MSE = deltaX2_sum / Etot;
     float X_RMS = sqrt(X_MSE);
     return X_RMS;
+  }
+   template <typename Delta = std::minus<float>>
+   float varXX(const std::vector<pair<float, float>>& energy_X_tc, const float X_cluster) const {
+    Delta delta;
+    float Etot = 0;
+    float deltaX2_sum = 0;
+    for (const auto& energy_X : energy_X_tc) {
+      deltaX2_sum += energy_X.first * pow(delta(energy_X.second, X_cluster), 2);
+      Etot += energy_X.first;
+    }
+    float X_MSE = 0;
+    if (Etot > 0)
+      X_MSE = deltaX2_sum / Etot;
+    return X_MSE;
```

Comment:
Should avoid duplicated code. This is redundant with `sigmaXX`.
What I would propose is to keep only `varXX` and remove `sigmaXX`. Implementations of `sigmaXX` would just take the `sqrt` of `varXX`, e.g. `sigmaPhiPhi` would return `sqrt(varPhiPhi)`